### DISCUSSION TITLE
How to get full path with locale using useRouter

### DISCUSSION THREAD

**Discussion by kcubero27** [vercel/next.js#30131] [Help]
For SEO purposes, I would like to **get the current URL including the locale** (if it's on the URL) using Static Generation.

Unfortunately, looking into [next/router documentation](https://nextjs.org/docs/api-reference/next/router) I realized that there isn't any method or property that returns the current URL with the locale and there is no way to detect on build time if the page includes or not the locale.

Imagine that we are on the page example.com/contact. In this case, I would like to receive /contact. If we are on example.com/en/contact, it would be en/contact and so on.

Is there any way to get the full path? 

Thank you in advance! 🙏🏻

**Comment by dalmo3** [comment]
You can get the locale at build time from `getStaticProps`'s context: https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation

**Reply by kcubero27** [reply]
I think I didn't explain myself.

What I want is a method from useRouter or something similar that gives me the entire URL with and without locale. That said, if I'm on the page /contact, I want to get /contact.

**Reply by dalmo3** [reply]
`window.location` wouldn't do?

**Reply by kcubero27** [reply]
I don't think that would work as it's at build time due to the static generation. With window I could only add it on CSR and I would like that to be at build time/SSR.

**Reply by dalmo3** [reply]
Then passing it as a prop from `getStaticProps` all the way to your component is the only way I can think of.

**Reply by kcubero27** [reply]
Hi Dalmo, I appreciate your efforts. But as I have just said, I don't need the locale, I need the full URL and that's not available in the getStaticProps method. Thanks anyway

**Reply by dalmo3** [reply]
OK. Starting over since you changed the original message.

If you want the full URL, you can't get it since next won't know what domain you're deploying to.

If you just want the pathname (e.g. /contact or /en/contact) you can definitely get it inside getStaticProps, since it's literally the same as the file path (except for `index` files), and you can access dynamic paths from `context.params` and `context.locale`, then you just do simple string concat and replacement.

**Reply by kcubero27** [reply]
@dalmo3 I think still you don't get what I'm saying here. I know you can get the pathname and the locale, but that's not what I'm saying. Please take a look into the example below so that you will understand it better.

Some examples of what I'm trying to accomplish:
URL: example.com/contact -> RESULT: /contact
URL: example.com/en/contact -> RESULT:  /en/contact
URL: example.com/de/contact -> RESULT:  /de/contact

As you can see, that's not as easy as putting together the pathname and the locale, because if you do that and your solutions you will get:
URL: example.com/contact -> RESULT: /en/contact
URL: example.com/en/contact -> RESULT:  /en/contact
URL: example.com/de/contact -> RESULT:  /de/contact

Why? Mainly because of the locale. It's never null when you are on the default locale and the path without locale. That's what I'm saying, THERE IS NO WAY TO GET THE URL WITHOUT THE LOCALE. Is it clear now?

**Reply by dalmo3** [reply]
I see it now, thanks. 

Are you sure the url `example.com/contact` exists at build time? If that's the case it looks like the official documentation is wrong:

> For Automatically Statically Optimized and non-dynamic getStaticProps pages, a version of the page will be generated for each locale. This is important to consider because it can increase build times depending on how many locales are configured inside getStaticProps.

> For example, if you have 50 locales configured with 10 non-dynamic pages using getStaticProps, this means getStaticProps will be called 500 times. 50 versions of the 10 pages will be generated during each build.

[Source](https://nextjs.org/docs/advanced-features/i18n-routing#dynamic-routes-and-getstaticprops-pages)

If what you describe is correct, the example above would generate 60 pages, with 10 additional pages without any locale in the path.
